### ISSUE TITLE
Screen reader doesn't announce question name and help text for checkbox, radio and enumerator questions

### ISSUE THREAD

**Issue by melanie-exygy** [civiform/civiform#10219]
**Describe the bug**
When a screen reader user is navigating through the form using "Tab" or other form controls, the name and help text isn't read on checkbox, radio and enumerator questions.  

**To Reproduce**

1. Apply to the Comprehensive Sample Program.
2. Turn the screen reader on.
3. Tab to the "Which of the following kitchen instruments do you own?" question.
4. Notice that the screen reader doesn't read the question or the help text.  It just says "Toaster".

**Expected behavior**
The question and help text should be read before the label of the first checkbox.

**Screenshots**

https://github.com/user-attachments/assets/bb159e15-b1ec-4725-bbca-4e9be493a2db

**Desktop (please complete the following information):**

- Device: Macbook
- Browser:  Chrome
- Screen reader: VoiceOver

**Additional context**
This is because the label and help text aren't programmatically associated with the input element on those questions.  For the question name, I believe the problem is that the `<legend>` is not the first child of the `<fieldset>`.  As for the help text, there is an `aria-describedBy` and that should be sufficient, but it may not be in the correct order.  See other question types for an example of what does work.

![Image](https://github.com/user-attachments/assets/3f9f1825-5725-4969-bcc1-239b61705889)

**Comment by tallulahkay** [comment]
@melanie-exygy the expected behavior that you specified is "The question and help text should be read before the label of the first checkbox." but the [uswds example](https://designsystem.digital.gov/components/checkbox/) has the question and help text read after the label of the first checkbox - do you think that's sufficient?

**Comment by melanie-exygy** [comment]
> [@melanie-exygy](https://github.com/melanie-exygy) the expected behavior that you specified is "The question and help text should be read before the label of the first checkbox." but the [uswds example](https://designsystem.digital.gov/components/checkbox/) has the question and help text read after the label of the first checkbox - do you think that's sufficient?

Yes, as long as the question and help text are read at some point, then I think it's fine.
